### PR TITLE
[sui-json-rpc] Fix byte generation for dev inspect

### LINKED ISSUE #8119: [bug] sui_devInspectTransaction failing on 0.24.0 devnet
In order to accommodate the new input to the subject RPC call, I have BCS serialized the SingleTransactionKind(PaySui) whichi produces the following bytes (put in u8 list for convenience):
```rust
[0, 5, 1, 20, 87, 32, 11, 222, 234, 36, 110, 94, 59, 37, 198, 115, 35, 47, 235, 70, 8, 226, 156, 93, 176, 2, 0, 0, 0, 0, 0, 0, 32, 155, 239, 255, 8, 3, 94, 134, 142, 215, 65, 22, 72, 140, 168, 45, 239, 22, 80, 186, 104, 63, 8, 70, 178, 190, 97, 159, 100, 116, 173, 41, 52, 1, 20, 212, 120, 254, 173, 89, 93, 24, 147, 228, 177, 174, 173, 38, 105, 189, 176, 159, 224, 227, 15, 1, 64, 66, 15, 0, 0, 0, 0, 0]
```

And the base64 string from above bytes:
```bash
AAUBFFcgC97qJG5eOyXGcyMv60YI4pxdsAIAAAAAAAAgm+//CANeho7XQRZIjKgt7xZQumg/CEayvmGfZHStKTQBFNR4/q1ZXRiT5LGurSZpvbCf4OMPAUBCDwAAAAAA
```

But I am getting:
`Inspect failure -> {'code': -32000, 'message': 'invalid length 0, expected an byte array of size 32'} `

There are more details in [forum](https://forums.sui.io/t/bcs-transactionkind-for-sui-devinspecttransaction-paysui/4776)

Also, if I just take the tx_bytes result of calling `sui_paySui` and feed the tx_bytes from that into devInspectTransaction, it fails as well.

### PR DESCRIPTION
**Pull request by tnowacki**
- incorrect bytes generated for dev-inspect Move call, where the full transaction data was serialized

Should fix #8119

### CODE REVIEW

**Comment by tnowacki** [comment]
shoot, let me try to fix my cargo fmt settings

File: crates/sui-transaction-builder/src/lib.rs [hunk 1]
```diff
@@ -524,7 +530,7 @@ impl<Mode: ExecutionMode> TransactionBuilder<Mode> {
         ))
     }
 
-    pub async fn batch_transaction(
+    pub async fn batch_transaction<Mode: ExecutionMode>(
```
Comment (666lcz) [inline crates/sui-transaction-builder/src/lib.rs:533]: Don't we only support `ExecutionMode::Normal` for batch transactions?

File: crates/sui-transaction-builder/src/lib.rs [hunk 2]
```diff
@@ -260,8 +258,14 @@ impl<Mode: ExecutionMode> TransactionBuilder<Mode> {
         gas_budget: u64,
     ) -> anyhow::Result<TransactionData> {
         let single_move_call = SingleTransactionKind::Call(
-            self.single_move_call(package_object_id, module, function, type_args, call_args)
-                .await?,
+            self.single_move_call::<execution_mode::Normal>(
```
Comment (666lcz) [inline crates/sui-transaction-builder/src/lib.rs:261]: I thought we don't want to leak `execution_mode` to transaction builder (https://github.com/MystenLabs/sui/pull/7236/files), is there anything changed?

Comment (tnowacki) [reply crates/sui-transaction-builder/src/lib.rs:261]: I do not understand why you wouldn't want to leak it to the builder. You would want to be able to specify why set of rules to run, no?

File: crates/sui-transaction-builder/src/lib.rs (hunk 1 above)
Comment (tnowacki) [reply crates/sui-transaction-builder/src/lib.rs:533]: We support batch in dev inspect IIRC

**Comment by tnowacki** [comment]
@patrickkuo 
Test fails here and not exactly sure what to do https://github.com/MystenLabs/sui/blob/bf589ac374aafd13a7df717807bce7dc2a22666c/crates/sui-json-rpc/tests/routing_tests.rs#L82

**Pull request closed without merging**